creator/data_templates/mutations/data_template.py

```diff
@@ -0,0 +1,207 @@
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.conf import settings
+
+from creator.organizations.models import Organization
+from creator.data_templates.models import DataTemplate
+from creator.data_templates.nodes.data_template import DataTemplateNode
+
+
+class CreateDataTemplateInput(graphene.InputObjectType):
+    """ Parameters used when creating a new data_template """
+
+    name = graphene.String(
+        required=True, description="The name of the data_template"
+    )
+    description = graphene.String(
+        required=True, description="The description of the data_template"
+    )
+    icon = graphene.String(
+        description="Name of the Font Awesome icon to use when displaying the "
+        "template"
+    )
+    organization = graphene.ID(
+        required=True,
+        description="The organization that will own this data_template"
+    )
+
+
+class UpdateDataTemplateInput(graphene.InputObjectType):
+    """ Parameters used when updating an existing data_template """
+
+    name = graphene.String(description="The name of the data_template")
+    description = graphene.String(
+        description="The description of the data_template"
+    )
+    icon = graphene.String(
+        description="Name of the Font Awesome icon to use when displaying the "
+        "template"
+    )
+    organization = graphene.ID(
+        description="The organization that will own this data_template"
+    )
+
+
+class CreateDataTemplateMutation(graphene.Mutation):
+    """ Creates a new data_template """
+
+    class Arguments:
+        input = CreateDataTemplateInput(
+            required=True, description="Attributes for the new data_template"
+        )
+
+    data_template = graphene.Field(DataTemplateNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new data_template.
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.add_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        # Check if organization exists
+        org_id = input.pop("organization")
+        model, node_id = from_global_id(org_id)
```

Comment:
Small nitpick on semantics: the global id is the same as the node's id. Really what you're getting back here using `from_global_id` is the node type and the primary identifier.